### PR TITLE
store client: add safe guards for when environment_auth is set (CRAFT-648)

### PR DESCRIPTION
**Pull request by sergiusens**
Login and storing the credentials to and environment variable or
logout to clear an environment variable should not disallowed.

Signed-off-by: Sergio Schvezov <sergio.schvezov@canonical.com>

-------

- [x] Have you signed the [CLA](http://www.ubuntu.com/legal/contributors/)?

-----

### CODE REVIEW

File: craft_store/store_client.py [hunk 1]
```diff
@@ -98,6 +99,9 @@ def __init__(
         self._endpoints = endpoints
 
         self._auth = Auth(application_name, base_url, environment_auth=environment_auth)
+        self._ephemeral_login = (
+            environment_auth is not None and os.getenv(environment_auth) is not None
```
Comment (cmatsuoka) [inline craft_store/store_client.py:103]: If `environment_auth` is set to an empty string or something else that's not a valid environment variable name we get `None` so we're good here.

**Review by cmatsuoka — APPROVED** [review]
Looks good.

File: craft_store/store_client.py [hunk 2]
```diff
@@ -139,6 +143,7 @@ def login(
         ttl: int,
         packages: Optional[Sequence[endpoints.Package]] = None,
         channels: Optional[Sequence[str]] = None,
+        save_credentials=True,
```
Comment (facundobatista) [inline craft_store/store_client.py:146]: Is there a case where login should be called when using auth from an env var?

**Review by facundobatista — CHANGES_REQUESTED** [review]

File: craft_store/store_client.py (hunk 2 above)
Comment (sergiusens) [reply craft_store/store_client.py:146]: I don't understand if you are asking for a change or asking a question. I would say yes for this library and no for Charmcraft specifically.

**Comment by sergiusens** [comment]
Abandoned

**Pull request closed without merging**